rifle.py

```diff
@@ -26,28 +27,61 @@ def __init__(self, universe, DRAWING_SCALE):
         self.current_text_ = ""
         self.enemy_color_ = RED
         self.font_size_ = 15
+        self.ids_for_target_tags_ = dict() 
         self.main_character_color_ = GREEN
-        self.target_tags_ = dict()
+        self.projectile_color_ = WHITE 
         self.target_tag_y_spacing_ = 5
+        self.target_tags_ = dict()
         self.targeting_terminal_ = TargetingTerminal(DRAWING_SCALE)
         self.text_antialias_ = 1
         self.ui_font_ = pygame.font.SysFont("monospace", self.font_size_*DRAWING_SCALE)
         self.word_generator_ = WordGenerator()
         self.word_length_min_ = 3
-        self.word_length_range = 3
-
-
+        self.word_length_range_ = 3
+        
+    def get_target_id(self,terminal_input): 
+        for enemy in self.universe_.enemies(): 
+            if terminal_input == self.target_tags_[enemy.ID_]:
+                target_ID = self.ids_for_target_tags_[terminal_input]
+                break
+            else:       
+                target_ID = None
+        return target_ID  
+
+    def check_if_input_matches_target_tag(self): 
+        target_ID =  self.get_target_id(self.current_text_)
+        if (target_ID != None): 
+            target_position = self.universe_.enemies_[target_ID].position_  
+            return target_position 
+        else:
+            return None
+    
     """
     Update Functions
     """
-
     def update(self, events):
         for enemy in self.universe_.enemies():
             if enemy.ID_ not in self.target_tags_:
-                self.target_tags_[enemy.ID_] = \
-                            self.word_generator_.request_word(self.word_length_min_, self.word_length_range)
-
-        self.targeting_terminal_.update(events)
+                new_word = self.word_generator_.request_word(self.word_length_min_, self.word_length_range_)
+                self.target_tags_[enemy.ID_] = new_word
+                self.ids_for_target_tags_[new_word] = enemy.ID_ 
+
+        for event in events:
+            if event.type == pygame.KEYDOWN:
+                if event.key == EVENT_KEY_ENTER:
+                    target_location = self.check_if_input_matches_target_tag()
+                    self.current_text_ = ""
+                    if target_location:
+                        rifle_projectile = RifleProjectile(self.universe_.main_character_.position_, target_location)
+                        self.universe_.create_friendly_projectile(rifle_projectile)
+                    
+                if event.key == EVENT_KEY_BACKSPACE:
+                    self.current_text_ = self.current_text_[:-1]
+                elif event.key in range(EVENT_KEY_a, EVENT_KEY_z+1):
+                    if len(self.current_text_)<= self.targeting_terminal_.max_word_size_:
+                        self.current_text_ += event.unicode.lower()
```

Comment:
This line was updated in previous code to be:
`self.current_text_ += chr(event.key)` to fix a bug. It didn't get caught in the merge due to the fix being in a separate file, do make sure this bug fix gets in though.